src/Symfony/Component/Lock/Tests/Store/ZookeeperStoreTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\StoreFactory;
+use Symfony\Component\Lock\Store\ZookeeperStore;
+
+/**
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ *
+ * @requires extension zookeeper
+ */
+class ZookeeperStoreTest extends AbstractStoreTest
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @return ZookeeperStore
+     */
+    public function getStore(): ZookeeperStore
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOSTT').':2181';
+
+        try {
+            $zookeeper = new \Zookeeper(implode(',', array($zookeeper_server)));
+        } catch (\ZookeeperConnectionException $ex) {
+            self::markTestSkipped($ex->getMessage());
+        }
+
+        return StoreFactory::createStore($zookeeper);
+    }
+
+    public function testSaveSucceedsWhenPathContainsMoreThanOneNode()
+    {
+        $store = $this->getStore();
+        $resource = '/baseNode/lockNode';
```

Comment:
Nice catch @jderusse on empty string. I made changes to accommodate that case. Leaving the tests in this class because `MemcachedStoreTest` seemed to be failing for some reason in 1 of the environment.